### PR TITLE
[AIRFLOW-2344] Fix `connections -l` to work with pipe/redirect

### PR DESCRIPTION
**Pull request by sekikn**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2344
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

    `airflow connections -l` uses 'tabulate' package with
    fancy_grid format, which outputs box drawing characters.
    It can occur UnicodeEncodeError with pipe or redirect,
    since the default encoding for Python 2.x is ascii.
    This PR fixes it and contains some flask8 related fixes.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Added tests.core:CliTests.test_cli_connections_list_redirect.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by Fokko** [comment]
Python 3 is unhappy :(

**Comment by sekikn** [comment]
Exactly. Revised PR :)

File: airflow/bin/cli.py [hunk 1]
```diff
@@ -1029,9 +1029,12 @@ def connections(args):
...
                               Connection.extra).all()
         conns = [map(reprlib.repr, conn) for conn in conns]
-        print(tabulate(conns, ['Conn Id', 'Conn Type', 'Host', 'Port',
+        msg = tabulate(conns, ['Conn Id', 'Conn Type', 'Host', 'Port',
                                'Is Encrypted', 'Is Extra Encrypted', 'Extra'],
-                       tablefmt="fancy_grid"))
+                       tablefmt="fancy_grid")
+        if sys.version_info[0] < 3:
+            msg = msg.encode('utf-8')
+        print(msg)
```
Comment (Fokko) [inline airflow/bin/cli.py:1037]: Some debugging leftovers ;)

Comment (sekikn) [reply airflow/bin/cli.py:1037]: That's not for debug, it just prints tabulated connections list! ;)

**Review by Fokko — APPROVED** [review]
Thanks @sekikn :)

**Pull request closed without merging**